### PR TITLE
Fathead validation unit test

### PR DESCRIPTION
**Pull request by jbarrett**
```
$ DDG_TEST_FATHEAD=perl_doc prove t/validate_fathead.t
t/validate_fathead.t .. Unicode duplicated on lines 2064, 4463 at t/lib/TestUtils.pm line 100.
AUTOLOAD duplicated on lines 2547, 7018 at t/lib/TestUtils.pm line 100.
and duplicated on lines 239, 11668 at t/lib/TestUtils.pm line 100.
Unix duplicated on lines 1748, 8598 at t/lib/TestUtils.pm line 100.
I18N::Langinfo duplicated on lines 251, 9699 at t/lib/TestUtils.pm line 100.
ASCII duplicated on lines 1012, 10170 at t/lib/TestUtils.pm line 100.
t/validate_fathead.t .. 1/? -X duplicated on lines 2879, 11509 at t/lib/TestUtils.pm line 100.

#   Failed test 'Checking for duplicate titles'
#   at t/validate_fathead.t line 13.
Titles missing coverage in output : $^a, $^c, $^child_error_native, $^d, $^e, $^encoding, $^f, $^h, $^i, $^l, $^m, $^n, $^o, $^open, $^p, $^r, $^re_debug_flags, $^re_trie_maxbuf, $^s, $^t, $^taint, $^unicode, $^utf8locale, $^v, $^w, $^warning_bits, $^wide_system_calls, $^x, $accumulator, $arg, $argv, $basetime, $child_error, $compiling, $debugging, $effective_group_id, $effective_user_id, $egid, $errno, $euid, $eval_error, $exceptions_being_caught, $executable_name, $extended_os_error, $format_formfeed, $format_line_break_characters, $format_lines_left, $format_lines_per_page, $format_name, $format_page_number, $format_top_name, $gid, $inplace_edit, $input_line_number, $input_record_separator, $last_match_end, $last_paren_match, $last_regexp_code_result, $list_separator, $match, $multiline_matching, $nr, $ofmt, $ofs, $ors, $os_error, $osname, $output_auto_flush, $output_field_separator, $output_record_separator, $perl_version, $perldb, $pid, $postmatch, $prematch, $process_id, $program_name, $real_group_id, $real_user_id, $rs, $subscript_separator, $subsep, $system_fd_max, $uid, $warning, %^h, %env, %inc, %overload, %sig, -b, -g, -k, -o, -r, -z, @argv, @inc, @last_match_start, __data__, __end__, __file__, __line__, __package__, begin, check, core, end, init, unitcheck at t/lib/TestUtils.pm line 108.

#   Failed test 'Testing language feature coverage'
#   at t/validate_fathead.t line 17.
# Looks like you failed 2 tests of 2.
t/validate_fathead.t .. Dubious, test returned 2 (wstat 512, 0x200)
Failed 2/2 subtests 

Test Summary Report
-------------------
t/validate_fathead.t (Wstat: 512 Tests: 2 Failed: 2)
  Failed tests:  1-2
  Non-zero exit status: 2
Files=1, Tests=2,  1 wallclock secs ( 0.02 usr  0.00 sys +  0.78 cusr  0.04 csys =  0.84 CPU)
Result: FAIL
```

Cc: @MariagraziaAlastra, @moollaza

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by russellholt** [comment]
Who should review this?

**Comment by jbarrett** [comment]
Probably @jdorweiler. Just wrapping up the disambiguation validation and nixing category validation (genuinely not much we can do with output.txt and categories)

**Comment by russellholt** [comment]
is this WIP then?

File: lib/fathead/perl_doc/cover/functions.txt [hunk 1]
```diff
@@ -0,0 +1,275 @@
+-A
```
Comment (GuiltyDolphin) [inline lib/fathead/perl_doc/cover/functions.txt]: Are these files supposed to be included? Are you planning to remove before merge?

Comment (jbarrett) [inline lib/fathead/perl_doc/cover/functions.txt]: Yep, they're basically test load for the tests - no plan to remove them.

Is there an issue with this?

I'm updating fathead docs to detail the intent and functions of these files.

Comment (GuiltyDolphin) [inline lib/fathead/perl_doc/cover/functions.txt]: @jbarrett Fair enough! So I'm guessing these are generated elsewhere then hard-coded into the repo? Ping me when you've got a ready draft for the docs, I'd be interested to read some of the nitty-gritty details.

Comment (jbarrett) [inline lib/fathead/perl_doc/cover/functions.txt]: Yep, you find an independent source for keywords, add them to the cover/ directory.

They're split into entities like functions, operators etc. in order to test for or perhaps generate a common set of redirects.

**Comment by moollaza** [comment]
@jbarrett Thanks! This LGTM :+1:n

**Pull request merged**